src/io_threads.c

```diff
@@ -387,3 +391,101 @@ int trySendWriteToIOThreads(client *c) {
     IOJobQueue_push(jq, ioThreadWriteToClient, c);
     return C_OK;
 }
+
+/* Internal function to free the client's argv in an IO thread. */
+void IOThreadFreeArgv(void *data) {
+    robj **argv = (robj **)data;
+
+    for (int i = 0;; i++) {
+        robj *o = argv[i];
+        if (o == NULL) {
+            continue;
+        }
+
+        int allocator_id = o->allocator_id;
+        decrRefCount(o);
+
+        /* The allocator_id is set to 0 to indicate that this is the last argument to free */
+        if (allocator_id == 0) {
+            break;
+        }
+    }
+
+    zfree(argv);
+}
+
+/* This function attempts to offload the client's argv to an IO thread.
+ * Returns C_OK if the client's argv were successfully offloaded to an IO thread,
+ * C_ERR otherwise. */
+int tryOffloadFreeArgvToIOThreads(client *c) {
+    if (server.active_io_threads_num <= 1 || c->argc == 0) {
+        return C_ERR;
+    }
+
+    size_t tid = (c->id % (server.active_io_threads_num - 1)) + 1;
+
+    IOJobQueue *jq = &io_jobs[tid];
+    if (IOJobQueue_isFull(jq)) {
+        return C_ERR;
+    }
+
+    int last_arg_to_free = -1;
+
+    /* Prepare the argv */
+    for (int j = 0; j < c->argc; j++) {
+        if (c->argv[j]->refcount > 1 || tid != c->argv[j]->allocator_id) {
+            decrRefCount(c->argv[j]);
+            /* Set argv[j] to NULL to avoid double free */
+            c->argv[j] = NULL;
+        } else {
+            last_arg_to_free = j;
+        }
+    }
+
+    /* If no argv to free, free the argv array at the main thread */
+    if (last_arg_to_free == -1) {
+        zfree(c->argv);
+        return C_OK;
+    }
+
+    /* We set the allocator_id of the last arg to free to 0 to indicate that
+     * this is the last argument to free. With this approach, we don't need to
+     * send the argc to the IO thread and we can send just the argv ptr. */
+    c->argv[last_arg_to_free]->allocator_id = 0;
```

Comment:
I think we should consider the benefit of using the allocatorId. For one it will probably not work in different scenarios (io/threads number changes, blocking commands etc...) but second I am not sure the benefit will be that high (by avoid arena contention) and just the free offloading. maybe we can have a different PR introducing this as another improvement?

Reply:
We could consider making lazyfree always enabled. Are there any drawbacks of lazyfree?

Reply:
We plan to do this as part of a separate PR. Please see https://github.com/valkey-io/valkey/pull/763. It will require some changes to the defrag and eviction code-flow/tests.

Reply:
OK, but if we do that, then maybe it's not worth the extra complexity of tracking the allocating thread? That's the question IMO.

Reply:
I don't thing that encoding the specific thread ID is more complex than flagging an object to indicate whether it was allocated by the main thread or by the I/O threads.

Reply:
I do not have a strong pro or con to add. I only think this can be an incremental built after we simplify this PR a bit.

Reply:
I had the other concern of if this helps at all if we aren't using the jemalloc allocator. I guess I would vote let's split the allocator ID out of this PR for now and merge the stuff that has broader consensus, and we can followup and understand the pinning the free to to the specific core in more detail.

Reply:
I removed the allocator ID for now. We can discuss it as part of the next PRs.